QuizApp/Realm/RealmDataSource.swift

```diff
@@ -0,0 +1,60 @@
+import Foundation
+import RealmSwift
+
+protocol RealmDataSourceProtocol {
+
+    func getAllQuizzes() -> [QuizDataModel]
+
+    func getQuizzes(for category: String) -> [QuizDataModel]
+
+    func saveQuizzes(_ quizzesObjModel: [QuizDataModel])
+
+}
+
+class RealmDataSource: RealmDataSourceProtocol {
```

Comment:
```suggestion
class QuizDatabaseDataSource: QuizDatabaseDataSourceProtocol {
```
You want to remove "Realm" from the name to make the specifics of the implementation abstract. If you were to change the DB now and switch to Core Data, ideally you want all external users of this protocol to be completely unphased and unaware of that change. 

On the other hand, we're adding Quiz as a prefix to the name, because we want to keep the domain of this class/protocol specific to one domain, in this case quizzes. 
If the app was suddenly introduced to a new domain use case, e.g. if you were to also store leaderboard info to the DB, you would create a new data source class/protocol specific to that domain (LeaderboardDatabaseDataSource) that would handle interactions with the database related to leaderboard objects, and that way you would have followed the separation of concerns principle again.